proposals/0030-core-image-story.md

```diff
@@ -0,0 +1,430 @@
+- `getImageService` (see [Image Services](#image-services))
+
+We choose `astro:assets` over `astro:image` on purpose, as to make it intuitive that more things might get exposed from there over time.
+
+## `src/assets`
+
+A new reserved folders for assets. Assets do not have to live in this folder, however putting them in this folder will unlock several benefits. Notably, we consider this folder to be a safe place for git-based CMSes to put any uploaded assets in.
+
+Through an included alias `~/assets` pointing to `src/assets`, it'll be easy for users to refer to any uploaded assets there.
+
+### Content Collections integration
+
+It is fairly common for one of the property of a Markdown piece of content to need to be a reference to an asset (think, cover image for an article, picture for an author etc).
+
+In tandem with the `src/assets` folder, we'd like to introduce a way for users to specify that a specific property needs to refer to a valid asset from the `src/assets` folder:
+
+```ts
+import { asset, defineCollection, z } from "astro:content";
+
+const blogCollection = defineCollection({
+  schema: z.object({
+    title: z.string(),
+    image: asset({ width: 1120, height: 1120 }),
+  }),
+});
+```
+
+Additionally, assets referred this way will be transformed automatically to the same shape as if the image was imported (see section below.), so they're ready to be used optimally. However, optimization / resizing is left to the user to do using the JavaScript API if they desire to.
+
+## New ESM shape for images imports
+
+Currently, importing images in Astro returns a simple `string` with the path of the image. The `@astrojs/image` integration and this RFC enhance this by instead returning the following shape: `{src: string, width: number, height: number, format: string}`.
+
+This shape has multiple benefits:
+
+- It allows users to easily construct `img` tags with no CLS, without needing to opt-in the entire `Image` pipeline if they don't require it:
+
+```astro
+---
+import image from "../my_image.png";
+---
+
+<img src={image.src} width={image.width} height={image.height} />
+```
+
+- This shapes gives all the information needed to build your own image integration. Depending on the transformer used, you'll often need the original metadata (ex: width and height) to resize the image while keeping the aspect ratio. Or, you might need a different encoder depending on the file format. With this output, all of this is given without needing further steps.
+
+## Image component
+
+A component that outputs a simple `<img> `tag with all the required attributes needed to show the image.
+
+### Properties
+
+#### width and height
+
+Those properties are used for resizing the image and ensuring there's no CLS. In most cases, the `width` and `height` would be automatically inferred from the source file and not passing them would just result in the same dimensions.
+
+The only time those properties are required is for remote images.
```

Comment:
Some other viable options, depending on the service: `width` and `aspectRatio` or `height` and `aspectRatio`.

Reply:
We purposely didn't go the `aspectRatio` route for this compared to `@astrojs/image` due to user confusion. You're definitely free to support it in your service or create a wrapper though